### PR TITLE
Allow opt out of code discipline

### PR DESCRIPTION
**Pull request by johanandren**
As decided during team meeting in Stockholm

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -427,14 +427,22 @@ Preferences > Editor > Code Style > Scala, select Scalafmt as formatter and enab
...
+### Code discipline opt out
+
+In addition to formatting the Akka build enforces code discipline through a set of compiler flags. While exploring ideas
+the discipline may be more of a hindrance than a help so it is possible to disable it by setting the system property `akka.no.discipline`
+to any non-empty string value when starting up sbt: `sbt -Dakka.no.discipline=youbet`
 
-- 2 spaces
-- `{` on same line as method name
-- in all other aspects, follow the [Oracle Java Style Guide](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html)
+PR validation includes the discpline flags and will fail if the sources are not compiling with them enabled so if toggling
```
Comment (johanandren) [inline CONTRIBUTING.md]: It's so undisciplined it's not even spelled correctly!

**Review by patriknw — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (renatocaval) [reply CONTRIBUTING.md]: I find the whole sentence confusing as well. 

Instead, we can say:

"PR validation includes the discipline flags and therefore may fail if the flags were disabled during development. Make sure you compile your code at least once with discipline enabled before sending a PR."

Comment (johanandren) [reply CONTRIBUTING.md]: Works for me

**Review by chbatey — APPROVED** [review]
I was thinking a task so you could do it without restarting but this is simpler

**Pull request merged**